tools/halide_malloc_trace.h

```diff
@@ -0,0 +1,230 @@
+#ifndef HALIDE_MALLOC_TRACE_H
+#define HALIDE_MALLOC_TRACE_H
+
+//---------------------------------------------------------------------------
+// The custom trace allocator can be used in an application by calling:
+//
+//   halide_enable_malloc_trace();
+//
+// The trace is off by default, compile with the following to turn it on:
+//
+//   -DHL_MEMINFO  Produce memory tracing in the custom malloc trace allocator
+//
+//   halide_malloc => [0x9e400, 0xa27ff], # size:17408, align:1K
+//   halide-header => [0x9e390, 0x9e3ff], # size:112, align:16
+//   halide_malloc => [0xa2880, 0xa6e9f], # size:17952, align:128
+//   halide-header => [0xa2820, 0xa287f], # size:96, align:32
+//   halide_free   => [0x9e390, 0x9e3ff], # size:112, align:16
+//   halide_free   => [0xa2820, 0xa287f], # size:96, align:32
+//
+//---------------------------------------------------------------------------
+#define HL_MEMBUFLEN  128
+
+extern "C" {
+
+extern void *malloc(size_t);
```

Comment:
Yes, this was originally in hexagon_standalone and the lightweight lw_* routines were to both avoid external dependencies as well as avoid any further heap allocation.  I will go ahead and replace the functionality using standard I/O libraries.